File: 0819-most-common-word/0819-most-common-word.py
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        paragraph = paragraph.lower()
        
        # 알파벳이 아닌 문자는 공백문자로 치환 후 split
        word_list = re.sub(r'[^a-zA-Z]', ' ', paragraph).split()
        
        # 'Hi. friend' 같은 경우 . 이 공백문자로 치환되면서
        # 연속된 공백이 발생한다.
        # 이는 split 시 빈 문자열을 발생시키므로 이를 제외한 리스트 생성
        new_word_list = [word for word in word_list if word != '']
        word_counter = collections.Counter(new_word_list)
        for word, _ in word_counter.most_common():
            if word in banned:
                continue
            else:
                return word

File: 0819-most-common-word/0819-most-common-word.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        paragraph = paragraph.replaceAll("[^A-Za-z0-9]+", " ").toLowerCase();
        List<String> ban_list = Arrays.asList(banned);
        Map<String, Integer> counter = new HashMap<>();
        for(String word: paragraph.split(" ")) {
            if (!ban_list.contains(word))
                counter.put(word, counter.getOrDefault(word, 0) + 1);
        }
        
        int max_count = 0;
        String answer = "";
        for (Map.Entry<String, Integer> entry: counter.entrySet()) {
            Integer value = entry.getValue();
            if (max_count < value) {
                answer = entry.getKey();
                max_count = entry.getValue();
            }
        }
        
        return answer;
    }
}